client/components/data/query-terms/index.jsx

```diff
@@ -27,7 +27,7 @@ class QueryTerms extends Component {
 	}
 
 	request( props ) {
-		if ( props.requesting ) {
+		if ( props.requesting || ! props.siteId ) {
```

Comment:
Alternatively, we could avoid rendering the component when the `siteId` isn't known. There was previously a desire to try to avoid accommodating invalid props in the query components since we'd had plans to use `<QueryPosts />` and `<QuerySites />` with falsey site ID to indicate a `/me/posts` or `/me/sites` request, though there's since been some discussion that this could be asking for trouble. See https://github.com/Automattic/wp-calypso/pull/7195#discussion_r73036019 .

Food for thought in any case.